commit f9f8b81b5d13f09595051a23badb8f4f2b559db3
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-04-28T00:52:21+00:00 (1 year, 5 months ago)
    
    joyent/bugview#24 Handle closing {noformat} tags at the end of the line

diff --git a/jirapub.js b/jirapub.js
index d3fb04c..54cc9a9 100644
--- a/jirapub.js
+++ b/jirapub.js
@@ -25,6 +25,16 @@ var CONFIG = read_config(LOG);
 
 var ALLOWED_DOMAINS = CONFIG.allowed_domains;
 
+var BLOCKS = [
+    '{noformat}',
+    '{code}',
+    '{panel}',
+    '{quote}',
+];
+
+var BLOCK_START_RE = new RegExp('^(' + BLOCKS.join('|') + ')(.)', 'gm');
+var BLOCK_END_RE = new RegExp('(.)(' + BLOCKS.join('|') + ')$', 'gm');
+
 var JIRA;
 var SERVER;
 
@@ -779,6 +789,10 @@ function
 format_markup(desc)
 {
 	var out = '';
+
+	desc = desc.replace(BLOCK_START_RE, '$1\n$3');
+	desc = desc.replace(BLOCK_END_RE, '$1\n$2');
+
 	var lines = desc.split(/\r?\n/);
 
 	var last_was_heading = false;
